Visitor.NET.AutoVisitableGen/Visitor.NET.AutoVisitableGen/AutoVisitableIncrementalSourceGenerator.cs

```diff
@@ -105,28 +119,63 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
 
             var namespaceName = classSymbol.ContainingNamespace.ToDisplayString();
 
-            var code = $@"// <auto-generated/>
-
-using Visitor.NET;
-
-namespace {namespaceName};
+            // last is closest
+            List<INamedTypeSymbol> containingTypes = new();
 
-public partial {typeKind.ToString().ToLower()} {visitableTypeName} :
-    IVisitable<{visitableTypeName}>
-{{
-    public override TReturn Accept<TReturn>(
-        IVisitor<{baseTypeName}, TReturn> visitor) =>
-        Accept(visitor);
-
-    public TReturn Accept<TReturn>(
-        IVisitor<{visitableTypeName}, TReturn> visitor) =>
-        visitor.Visit(this);
-}}
-";
+            for (INamedTypeSymbol? currentType = classSymbol.ContainingType; currentType != null; currentType = currentType.ContainingType)
+            {
+                containingTypes.Add(currentType);
+            }
+            
+            containingTypes.Reverse();
```

Comment:
this algorithm looks sophisticated in place.

that's not great, considering it is a core of nested types support.

i'd extract it to some external public dependency (public static class will be ok i guess) to make it testable

Then, of course, write the tests

Reply:
- Moved it to class named `SymbolsHelper`
- Wrote tests